### ISSUE TITLE
[Bug]: Widevine version is not supported anymore

### ISSUE THREAD

**Issue by heydarian** [mozilla-mobile/fenix#22899]
### Steps to reproduce

Like two weeks ago Amazon Prime Video was working without issues in Desktop Mode.
Now I always get this error, it seems not supported/ outdated now.
![prime](https://user-images.githubusercontent.com/35915266/146607159-cecad22d-a88c-4e33-abbf-af0efe391dab.png)

### Expected behaviour

Play Amazon Prime Video DRM content

### Actual behaviour

![prime](https://user-images.githubusercontent.com/35915266/146607222-22a0db9d-8286-4ba8-97eb-147b8e27bf87.png)

### Device name

Polestar 2

### Android version

AAOS 10

### Firefox release type

Firefox

### Firefox version

96

**Comment by cpeterso** [comment]
@heydarian - I see your Firefox version is 96. Is that Firefox Beta?

Do any of the Widevine videos on the test website below play for you? I wonder if Amazon changed something or if this is a general problem with all Widevine videos.

https://shaka-player-demo.appspot.com/demo/

**Comment by heydarian** [comment]
hi, the Test works fine, also other DRM streaming Sites Like Disney+ work. It seems that amazon has changed something which hinders the Playback now.
any clue. why?

this ist my nightly build Info:

96.0a1 (Build #2015843967)
AC: 96.0.20211103143130, 2a27a32498
GV: 96.0a1-20211102094141
AS: 86.0.1

**Comment by cpeterso** [comment]
I just tested Amazon Prime on my old Android 8.1 phone and I was able to play DRM video just fine. I'm asking the Firefox media engineers how to figure out which Widevine version you have. I'm surprised AAOS 10 would have an older Widevine version than Android 8.1. On desktop, Firefox installs a Widevine library itself. On Android, Firefox uses the Widevine library provided by the OS. Maybe this is not a version problem, but Amazon Prime now requiring some hardware DRM that is not in your car?

**Comment by heydarian** [comment]
That's interesting to know, I thought that Firefox has included DRM [since v85](https://uk.pcmag.com/browsers/131447/firefox-85-for-android-allows-drm-protected-content-to-play-again)

I just ran the [DRM Info app](https://play.google.com/store/apps/details?id=com.androidfung.drminfo&hl=de&gl=US) and it gave me this output, could you please post the result of your Android 8.1 phone?
<img width="384" alt="drm" src="https://user-images.githubusercontent.com/35915266/146671388-b1cbc9b5-557d-44ba-b097-3f9a4aa6515f.png">

**Comment by cpeterso** [comment]
> That's interesting to know, I thought that Firefox has included DRM [since v85](https://uk.pcmag.com/browsers/131447/firefox-85-for-android-allows-drm-protected-content-to-play-again)

Firefox on Android didn't install a DRM library like it does on desktop. It just uses the DRM library (Google's Widevine DRM) that's already on the phone.

> I just ran the [DRM Info app](https://play.google.com/store/apps/details?id=com.androidfung.drminfo&hl=de&gl=US) and it gave me this output, could you please post the result of your Android 8.1 phone?

Here's my DRM info. I think the difference might be "Security Level": I have L1 and you have L3.

![image](https://user-images.githubusercontent.com/629193/146689214-0e756aad-65aa-4a5e-aa6a-6b97e4a34265.png)

**Comment by heydarian** [comment]
Hi again, could you please try this app also, which provides a more detailed info on your Widevine version:
https://play.google.com/store/apps/details?id=com.kaltura.kalturadeviceinfo&hl=en_AU&gl=US

<I will add my information later in here>

**Comment by cpeterso** [comment]
@heydarian - Does Amazon Prime work in Chrome on your device? Is there an official Amazon Prime app for your device?

My phone's Security Level L1 is more "secure" (more DRM-y) than your L3, so Amazon might be requiring more restrictive DRM. Mozilla has contacts at Amazon, so I can ask them once I know whether Amazon Prime works in Chrome and/or the Amazon Prime app on your device.

You said the problem started about two weeks before you filed this issue, so that would be around December 6. If you can recall a more precise date, that might narrow the regression window for our Amazon contacts to identify which code change caused the problem.

**Comment by heydarian** [comment]
@cpeterso there's officially no chrome or Amazon Prime App available for Android Automotive OS, see the current apps available here: https://play.google.com/store/apps/collection/promotion_android_auto_embedded_exploreall
I got Firefox by building it from Github and adding it to my google play developer console as internal app, so I can download internal apps back on my Android Automotive Google Playstore Account (USB-Debugging/ ADB is disabled in AAOS). For Chrome, there's only chromium available in Github and Amazon Prime Video isn't available at all (non-open source). However I tested the full chrome on the AAOS 10 emulator (USB-Debugging/ ADB is enabled there) on my PC and there Amazon Prime Video starts playing but with audio only (and black screen).
The last time, I player Amazon Prime successfully in Firefox in AAOS was on the 4th of November, since then I haven't tried it again until the 6th of December. So within those 4 weeks something changed at Amazon, as I always kept the same Firefox build (also tried the latest one from Github, no change).

Thanks for you help,
Darius

**Comment by heydarian** [comment]
Hi again,
@cpeterso today the Vivaldi Browser has been released and Amazon Prime Works!
https://play.google.com/store/apps/details?id=com.polestar.vivaldi.production.android

We can close this topic for now

**Comment by cpeterso** [comment]
> I got Firefox by building it from Github and adding it to my google play developer console as internal app, so I can download internal apps back on my Android Automotive Google Playstore Account (USB-Debugging/ ADB is disabled in AAOS).

Cool! I'm impressed by your effort and patience to install Firefox. 🙂

Did you need to make any code changes to Firefox to run in on AAOS? Is the AAOS Play Store a separate store or just a "This app is AAOS-compatible" flag that needs to be set in the regular Google Play Store?

>  @cpeterso today the Vivaldi Browser has been released and Amazon Prime Works!

Interesting. I wonder why Amazon Prime works in Vivaldi, but not Fenix or even Chrome.

**Comment by heydarian** [comment]
Hi @cpeterso, no code change required I was able to build FF right away with Android Studio.
The AAOS Play Store is a separate store, but as far as I have tested, by rebuilding open source apps, many work 1:1 without any limitations, the OS seems to be fully Android 10 compatible.
About Vivaldi, I think that Chrome would have worked the same in the real car environment as the have the same core. 
I only failed to play the video in the AAOS emulator on my PC.
